File: math/069-sqrt-x/SqrtX.java
class Solution {
    public int mySqrt(int x) {
        int left = 1, right =x;
        int ans = 0;
        while(left<=right){
            int mid = left + (right-left)/2;
            if((long)mid*mid==x){
                return mid;
            }
            else if((long)mid*mid<x){
                ans = mid;
                left = mid+1;
            }else{
                right = mid-1;
            }
        }
        return ans;
    }
}

File: math/069-sqrt-x/SqrtX.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 1, x
        ans = 0
        while left<=right:
            mid = left + (right - left)//2
            if mid*mid == x:
                return mid
            elif mid*mid<x:
                ans = mid
                left = mid+1
            else:
                right = mid-1
        return ans